internal/codegen/golang/gen.go

```diff
@@ -180,6 +189,12 @@ func generate(req *plugin.CodeGenRequest, options *opts, enums []Enum, structs [
 		"emitPreparedQueries": tctx.codegenEmitPreparedQueries,
 		"queryMethod":         tctx.codegenQueryMethod,
 		"queryRetval":         tctx.codegenQueryRetval,
+		"dollar": func() bool {
+			return req.Settings.Engine == string(config.EnginePostgreSQL)
+		},
+		"hasDynamic": func() bool {
```

Comment:
I wasn't sure how to add these values without putting them here. So while it does work it is kind of a hack.